app/main.py

```diff
@@ -1,3 +1,51 @@
+from typing import Any
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: int | float) -> object:
+        if isinstance(other, int | float):
+            distance2 = self.km + other
+            return Distance(distance2)
+        else:
+            distance3 = self.km + other.km
+            return Distance(distance3)
+
+    def __iadd__(self, other: int | float) -> object:
+        if isinstance(other, int | float):
+            self.km += other
+            return self
+        else:
+            self.km += other.km
+            return self
```

Comment:
You can rewrite it with one `return` statement. It makes your code more readable
```suggestion
        if isinstance(other, int | float):
            self.km += other
        else:
            self.km += other.km
        return self
```